Lib/pydoc.py

```diff
@@ -1494,8 +1571,11 @@ def docroutine(self, object, name=None, mod=None, cl=None):
         if name == realname:
             title = self.bold(realname)
         else:
-            if cl and inspect.getattr_static(cl, realname, []) is object:
-                skipdocs = 1
+            if (cl is not None and
+                inspect.getattr_static(cl, realname, []) is object):
```

Comment:
The default value here is rather strange. Why is it `[]`?

Reply:
Because it is the simplest method to create an unique not shared value. `[] is object` is always false, so this condition means "the class have such attribute and it is the same as object".

Reply:
ISTM like that piece of information should be recorded as a comment in the code.

Reply:
This is a common idiom. Well, usually it is used in more complex code:
```py
sentinel = [...]
result = getattr(obj, name, sentinel)
if result is sentinel: ...
```
But in this particular case no need to introduce variables for the result and the sentinel, so it all can be simplified.